src/ShaderPhong.h

```diff
@@ -48,27 +58,36 @@ class CShaderPhong : public CShaderFlat
 		shadow.org = ray.org + ray.t * ray.dir;
 
 		// iterate over all light sources
-		for (auto pLight : m_scene.getLights()) {
+		for (auto pLight : m_scene.m_vpLights) {
 			// get direction to light, and intensity
-			std::optional<Vec3f> lightIntensity = pLight->illuminate(shadow);
-			if (lightIntensity) {
-				// diffuse term
-				float cosLightNormal = shadow.dir.dot(normal);
-				if (cosLightNormal > 0) {
-					if (m_scene.occluded(shadow))
-						continue;
-
-					Vec3f diffuseColor = m_kd * color;
-					res += (diffuseColor * cosLightNormal).mul(lightIntensity.value());
-				}
-
-				// specular term
-				float cosLightReflect = shadow.dir.dot(reflect);
-				if (cosLightReflect > 0) {
-					Vec3f specularColor = m_ks * RGB(1, 1, 1); // white highlight;
-					res += (specularColor * powf(cosLightReflect, m_ke)).mul(lightIntensity.value());
-				}
+			for(int i = 0; i < nAreaSamples; i++) {
```

Comment:
Why did you incorporated area samples here ? Is it a copy-paste from last year assignments ?
In this assignment it is not needed at all.